### PR TITLE
[Sparse] [lang] Add ti.activate support

### PR DESCRIPTION
**Pull request by archibate**
Related issue = https://github.com/taichi-dev/taichi/pull/1256#discussion_r445026670

[[Click here for the format server]](http://kun.csail.mit.edu:31415/)

----
Would you write a test for me? I'm not super familiar with sparse.

### CODE REVIEW

File: tests/python/test_sparse_activate.py [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+def test_pointer():
+    ti.init(print_ir=True)
+
+    x = ti.var(ti.f32)
+    s = ti.var(ti.i32)
+
+    n = 16
+
+    ptr = ti.root.pointer(ti.i, n)
+    ptr.dense(ti.i, n).place(x)
```
Comment (k-ye) [inline tests/python/test_sparse_activate.py:14]: nit: Actually WDYT if we remove this `dense` layer, i.e. `ptr.place(x)`? If so, `s[None]` should be exactly how many elements in `x` that are activated. It makes the test more obvious?

File: tests/python/test_sparse_activate.py [hunk 2]
```diff
@@ -0,0 +1,34 @@
+import taichi as ti
+
+
+@ti.archs_support_sparse
+def test_pointer():
+    ti.init(print_ir=True)
+
```
Comment (yuanming-hu) [inline tests/python/test_sparse_activate.py:7]:
```suggestion
```

File: tests/python/test_sparse_deactivate.py [hunk 3]
```diff
@@ -9,7 +9,8 @@ def test_pointer():
     n = 16
 
     ptr = ti.root.pointer(ti.i, n)
-    ptr.dense(ti.i, n).place(x)
+    den = ptr.dense(ti.i, n)
+    den.place(x)
```
Comment (yuanming-hu) [inline tests/python/test_sparse_deactivate.py:13]: Any good reason to change this?

Comment (archibate) [reply tests/python/test_sparse_deactivate.py:13]:
```suggestion
    ptr.dense(ti.i, n).place(x)
```
Was testing, didn't revert, sorry!

File: tests/python/test_sparse_activate.py (hunk 1 above)
Comment (archibate) [reply tests/python/test_sparse_activate.py:14]: Maybe we want to test the behavior of multi-layer sparse, i.e. make OpenGL fake `dynamic` support fail to test.

**Pull request merged**